### PR TITLE
Study-Export should include Participants and Integrations

### PR DESCRIPTION
**Pull request by drtyyj**

### CODE REVIEW

File: studymanager/src/main/java/io/redlink/more/studymanager/model/StudyImportExport.java [hunk 1]
```diff
@@ -19,6 +20,8 @@ public class StudyImportExport {
     private List<Intervention> interventions;
     private Map<Integer, Trigger> triggers;
     private Map<Integer, List<Action>> actions;
+    private Map<String,Integer> participantGroupAssignments;
```
Comment (ja-fra) [inline studymanager/src/main/java/io/redlink/more/studymanager/model/StudyImportExport.java:23]: Why did you choose `<String, Integer>` and not `<Integer, Integer>` here?

File: studymanager/src/main/resources/openapi/StudyManagerAPI.yaml [hunk 2]
```diff
@@ -1783,6 +1783,23 @@ components:
...
+
+          additionalProperties:
+            type: integer
+            format: int32
+        integrations:
+          type: array
+          items:
+            $ref: '#/components/schemas/integrationInfo'
+
+    integrationInfo:
```
Comment (ja-fra) [inline studymanager/src/main/resources/openapi/StudyManagerAPI.yaml:1796]: Nitpicking...
```suggestion
            $ref: '#/components/schemas/IntegrationInfo'

    IntegrationInfo:
```

File: studymanager/src/main/java/io/redlink/more/studymanager/model/StudyImportExport.java (hunk 1 above)
Comment (ja-fra) [reply studymanager/src/main/java/io/redlink/more/studymanager/model/StudyImportExport.java:23]: As an alternative, we could also export the group-assignment as a simple `List<Integer>`.

For export, you would just do an
```
ParticipantService.listParticipants(studyId)
    .stream()
    .map(Participant::getStudyGroupId)
    .toList()
```

This way we would also keep the assignment-ordering...

File: studymanager/src/main/java/io/redlink/more/studymanager/service/ImportExportService.java [hunk 3]
```diff
@@ -116,6 +135,19 @@ public Study importStudy(StudyImportExport studyImport, AuthenticatedUser user)
                         studyImport.getActions().getOrDefault(intervention.getInterventionId(), Collections.emptyList())
                 )
         );
+        studyImport.getParticipantGroupAssignments().forEach((key, value) -> {
+            for (int i = 0; i < value; i++) {
+                participantService.createParticipant(
+                        new Participant()
+                                .setStudyId(studyId)
+                                .setAlias("Alias")
```
Comment (ja-fra) [inline studymanager/src/main/java/io/redlink/more/studymanager/service/ImportExportService.java:143]:
```suggestion
                                .setAlias("Participant")
```

File: studymanager/src/main/java/io/redlink/more/studymanager/model/StudyImportExport.java (hunk 1 above)
Comment (drtyyj) [reply studymanager/src/main/java/io/redlink/more/studymanager/model/StudyImportExport.java]: I couldn't figure out how to make openapi to generate the class in such a way, that the map is a Map<Integer, Integer>. And according to a stackoverflow post, it is not possible, as keys have to be Strings there. But i guess i should have transformed the Strings to Integers inside the Transformer.

**Review by ja-fra — APPROVED** [review]

**Pull request merged**